lib/srv/db/ca.go

```diff
@@ -59,13 +59,45 @@ func (s *Server) startCARenewer(ctx context.Context) {
 // initCACert initializes the provided server's CA certificate in case of a
 // cloud hosted database instance.
 func (s *Server) initCACert(ctx context.Context, database types.Database) error {
+	s.mu.RLock()
+	if !s.shouldInitCACertLocked(database) {
+		s.mu.RUnlock()
+		return nil
+	}
+	// make a copy so we can safely unlock before doing expensive CA cert
+	// version checking or downloading.
+	copy := database.Copy()
+	s.mu.RUnlock()
+	// It's not set so download it or see if it's already downloaded.
```

Comment:
This comment seems misplaced, could we either remove or fix it?

Reply:
this comment was there before, and as far as I can tell doesn't correspond to what actually happens regarding ".. do not update the certificates...", and the comment inside `shouldInitCACertLocked` better explains the part about the CA being set or not.

I'll just remove this comment.

Reply:
removed comment in [abab275](https://github.com/gravitational/teleport/pull/29215/commits/abab2753ed02582bc7bf9874c32c0b51bfb8de1d)

Reply:
also, the motivation for making the `shouldInitCACertLocked` helper func was to make it easier to visually see that we handle locking/unlocking correctly.